src/Libplanet.RocksDBStore/RocksDBStore.cs

```diff
@@ -1454,12 +1473,221 @@ public override void DeleteNextStateRootHash(BlockHash blockHash)
+                throw;
+            }
+            finally
+            {
+                _rwEvidenceLock.ExitWriteLock();
+            }
+        }
+
+        /// <inheritdoc/>
+        public override void DeletePendingEvidence(EvidenceId evidenceId)
+        {
+            byte[] key = PendingEvidenceKey(evidenceId);
+
+            if (!(_pendingEvidenceDb.Get(key) is { }))
+            {
+                return;
+            }
+
+            _rwEvidenceLock.EnterWriteLock();
+            try
+            {
+                _pendingEvidenceDb.Remove(key);
+            }
+            catch (Exception e)
+            {
+                LogUnexpectedException(nameof(DeletePendingEvidence), e);
+                throw;
+            }
+            finally
+            {
+                _rwEvidenceLock.ExitWriteLock();
+            }
+        }
+
+        /// <inheritdoc/>
+        public override bool ContainsPendingEvidence(EvidenceId evidenceId)
+        {
+            return _pendingEvidenceDb.Get(PendingEvidenceKey(evidenceId)) is { };
+        }
+
+        /// <inheritdoc/>
+        public override Evidence? GetCommittedEvidence(EvidenceId evidenceId)
+        {
+            if (_evidenceCache.TryGetValue(evidenceId, out Evidence cachedEvidence))
+            {
+                return cachedEvidence;
+            }
+
+            _rwEvidenceLock.EnterReadLock();
+
+            try
+            {
+                byte[] key = CommittedEvidenceKey(evidenceId);
+                byte[] bytes = _committedEvidenceDb.Get(key);
+                if (bytes is null)
+                {
+                    return null;
+                }
+
+                return Evidence.Decode(Codec.Decode(bytes));
+            }
+            catch (Exception e)
+            {
+                LogUnexpectedException(nameof(GetCommittedEvidence), e);
+            }
+            finally
+            {
+                _rwEvidenceLock.ExitReadLock();
+            }
+
+            return null;
+        }
+
+        /// <inheritdoc/>
+        public override void PutCommittedEvidence(Evidence evidence)
+        {
+            if (_evidenceCache.ContainsKey(evidence.Id))
+            {
+                return;
+            }
+
+            byte[] key = CommittedEvidenceKey(evidence.Id);
+
+            if (_committedEvidenceDb.Get(key) is { })
+            {
+                return;
+            }
+
+            _rwEvidenceLock.EnterWriteLock();
+            try
+            {
+                byte[] value = Codec.Encode(Evidence.Bencode(evidence));
+                _committedEvidenceDb.Put(key, value);
+                _evidenceCache.AddOrUpdate(evidence.Id, evidence);
+            }
+            catch (Exception e)
+            {
+                LogUnexpectedException(nameof(PutCommittedEvidence), e);
+                throw;
+            }
+            finally
+            {
+                _rwEvidenceLock.ExitWriteLock();
+            }
```

Comment:
Maybe we can store evidence separately with a block. like transactions.

Reply:
This issue has been postponed.